### PR TITLE
Fix incorrect profiling region naming in CrsGraph

### PR DESCRIPTION
**Pull request by aprokop**
When running ordered traversals with `CrsGraph`, it incorrectly printed `nearest` instead of `ordered_traversal` as a profiling region.

Drive-by change to use `constexpr` when labeling in `BVH`.

### CODE REVIEW

File: src/ArborX_LinearBVH.hpp [hunk 1]
```diff
@@ -239,15 +239,16 @@ void BasicBoundingVolumeHierarchy<MemorySpace, BoundingVolume, Enable>::query(
 
   using Tag = typename Details::AccessTraitsHelper<Access>::tag;
   std::string profiling_prefix = "ArborX::BVH::query";
-  if (std::is_same<Tag, Details::SpatialPredicateTag>{})
+  if constexpr (std::is_same_v<Tag, Details::SpatialPredicateTag>)
```
Comment (dalg24) [inline src/ArborX_LinearBVH.hpp:242]: Do not put unrelated (somewhat unnecessary) changes in the same commit.
Either revert or move into another commit with an appropriate message.
(And if the rational is more consistency, you might want to decide where you want the `::`, in the prefix to like here in what gets appended to it.

**Review by dalg24 — DISMISSED** [review]

**Review by dalg24 — APPROVED** [review]

File: src/ArborX_LinearBVH.hpp [hunk 2]
```diff
@@ -238,22 +238,23 @@ void BasicBoundingVolumeHierarchy<MemorySpace, BoundingVolume, Enable>::query(
...
+  else if constexpr (std::is_same_v<Tag,
+                                    Experimental::OrderedSpatialPredicateTag>)
   {
-    profiling_prefix += "::ordered_spatial";
+    profiling_prefix += "ordered_spatial";
   }
   else
   {
-    Kokkos::abort("ArborX: implementation bug");
+    static_assert(std::is_void_v<Tag>, "ArborX implementation bug");
```
Comment (dalg24) [inline src/ArborX_LinearBVH.hpp:257]: I would have preferred if you left the "ArborX" out.
```suggestion
    static_assert(std::is_void_v<Tag>, "implementation bug");
```

Comment (aprokop) [reply src/ArborX_LinearBVH.hpp:257]: Why?

Comment (dalg24) [reply src/ArborX_LinearBVH.hpp:257]: the "implementation bug" is more a hint in case someone is not familiar with this trick (because you can't just write `static_assert(false);`).  "ArborX" is pointless, any compiler would tell you that it was raised from an ArborX function.  And the key here is that it is not about diagnostic it is about readability of the code.

Comment (aprokop) [reply src/ArborX_LinearBVH.hpp:257]: I don't see how "implementation bug" is more readable than "ArborX implementation bug".
